src/messages/IPreMessageSentHandler.ts

```diff
@@ -22,13 +20,13 @@ export interface IPreMessageSentHandler {
      * @param read An accessor to the environment
      * @param extend An accessor for modifying the messages non-destructively
      */
-    extendMessage(message: IMessage, read: IMessageRead, extend: IMessageExtend): void;
+    extendMessage(message: IMessage, read: IRead, extend: IMessageExtend): void;
```

Comment:
@graywolf336 I believe the Read-Interface should always have a scope as wide as possible. It's much different with respect to modifying accessors to me though